csharp/src/Ice/IPEndpoint.cs

```diff
@@ -277,6 +300,19 @@ private protected IPEndpoint(InputStream istr, Protocol protocol)
                     Host = oaEndpoint ? "::0" :
                         throw new FormatException($"`-h *' not valid for proxy endpoint `{endpointString}'");
                 }
+
+                if (!oaEndpoint && IPAddress.TryParse(Host, out IPAddress? address))
+                {
+                    if (address.Equals(IPAddress.Any))
```

Comment:
For performance reasons, can we/should we defer this checking until we actually use this endpoint?

My concern is every time we unmarshal a direct proxy, we IPAddress-parse every single endpoint to make sure these endpoints are not 0.0.0.0/::0, which sounds like a waste.

Reply:
I think it's better to do it here than to get a bogus endpoint. Proxies (or any type for that matter) which are encoded incorrectly should fail to during unmarshaling.

Reply:
Yes, there is little chance this happens as we don't allow to construct such an endpoint, it will mean the endpoint was marshall by an old version or by hand. I think we can just remove it from the unmarshal constructor

Reply:
I remove this check for now, if we unmarshal IP any address it will fail on connection establishment